cmd/queue/main_test.go

```diff
@@ -443,6 +443,38 @@ func TestQueueTraceSpans(t *testing.T) {
 	}
 }
 
+func BenchmarkProxyHandler(b *testing.B) {
+	var httpHandler http.HandlerFunc = func(w http.ResponseWriter, r *http.Request) {}
+	server := httptest.NewServer(httpHandler)
+	serverURL, _ := url.Parse(server.URL)
+	defer server.Close()
+	proxy := httputil.NewSingleHostReverseProxy(serverURL)
+	params := queue.BreakerParams{QueueDepth: 10, MaxConcurrency: 10, InitialCapacity: 10}
+	breaker := queue.NewBreaker(params)
+	testFunc := func() {
+		reqChan := make(chan queue.ReqEvent, requestCountingQueueLength)
+		h := proxyHandler(reqChan, breaker, true /*tracingEnabled*/, proxy)
+		req := httptest.NewRequest(http.MethodPost, "http://example.com", nil)
+		req.Header.Set(network.OriginalHostHeader, wantHost)
+		resp := httptest.NewRecorder()
```

Comment:
Do you need a full recorder? It's going to record the responses body in memory, thus skewing the metrics of the benchmark. Check the activator's handler test for an alternative.